tests/wpunit/Admin/Fields/FieldTest.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+namespace StellarWP\Uplink\Admin\Fields;
+
+use StellarWP\Uplink\Admin\Group;
+use StellarWP\Uplink\Config;
+use StellarWP\Uplink\Register;
+use StellarWP\Uplink\Resources\Collection;
+use StellarWP\Uplink\Tests\UplinkTestCase;
+use StellarWP\Uplink\Tests\TestUtils;
+use tad\Codeception\SnapshotAssertions\SnapshotAssertions;
+use StellarWP\Uplink\Tests\Traits\With_Uopz;
+
+class FieldTest extends UplinkTestCase {
+
+	use TestUtils;
+	use SnapshotAssertions;
+	use With_Uopz;
+
+	public function resourceProvider() {
+		$resources = $this->get_test_resources();
+
+		foreach ( $resources as $resource ) {
+			yield $resource['slug'] => [ $resource ];
+		}
+	}
+
+	public function setup_container_get_slug( $resource ) {
+		$collection = Config::get_container()->get( Collection::class );
+
+		Register::{$resource['type']}(
+			$resource['slug'],
+			$resource['name'],
+			$resource['version'],
+			$resource['path'],
+			$resource['class'],
+		);
+
+		return $collection->get( $resource['slug'] );
+	}
+
+	/**
+	 * @test
+	 * @dataProvider resourceProvider
+	 */
+	public function it_should_get_fields_with_slug( $resource ) {
+		$current_resource = $this->setup_container_get_slug( $resource );
+		$slug             = $current_resource->get_slug();
+		$field            = $this->container->get( Field::class )->set_resource( $current_resource );
+		$license_key      = 'license_key' . $slug;
+		$option_name      = $current_resource->get_license_object()->get_key_option_name();
+
+		// Update the license key to a known value.
+		update_option( $option_name, $license_key );
+		$option_value = get_option( $option_name );
+		$this->assertEquals( $option_value, $license_key );
+
+		$field_name = 'field-' . $slug;
+		$field->set_field_name( $field_name );
+
+		// Add assertions to verify the field properties and methods
+		$this->assertEquals( $slug, $field->get_slug() );
+		$this->assertEquals( $current_resource->get_path(), $field->get_product() );
+		$this->assertEquals( $field_name, $field->get_field_name() );
+		$this->assertEquals( 'stellarwp_uplink_license_key_' . $slug, $field->get_field_id() );
+		$this->assertEquals( $license_key, $field->get_field_value(), 'Field value should be equal to the license key' );
+		$this->assertStringContainsString( 'A valid license key is required for support and updates', $field->get_key_status_html() );
+		$this->assertEquals( 'License key', $field->get_placeholder() );
+		$this->assertEquals( 'stellarwp-uplink-license-key-field', $field->get_classes() );
+	}
+
+	/**
+	 * @test
+	 * @dataProvider resourceProvider
+	 */
+	public function it_should_render_fields_correctly( $resource ) {
+		$current_resource = $this->setup_container_get_slug( $resource );
```

Comment:
If I'm correct, the current user is `0` (not logged in), should the field render for all users?

Reply:
If the user doesn't have the ability to access the field it will display the message `Contact your network administrator to connect`. 
https://github.com/stellarwp/uplink/blob/feat/fluent-form-builder/src/Uplink/Components/Admin/Authorize_Button_Controller.php#L110

Although similar to what @dpanta94 said above, it may be something we want to discuss.